lib/services/livesync/device-livesync-service-base.ts

```diff
@@ -0,0 +1,26 @@
+export abstract class DeviceLiveSyncServiceBase<T extends Mobile.IDevice> {
+	protected get device(): T {
+		return <T>(this._device);
+	}
+
+	constructor(private _device: Mobile.IDevice,
+		private $liveSyncProvider: ILiveSyncProvider) { }
+
+	public refreshApplication(deviceAppData: Mobile.IDeviceAppData, localToDevicePaths: Mobile.ILocalToDevicePathData[], forceExecuteFullSync: boolean): IFuture<void> {
+		let canExecuteFastSync = !forceExecuteFullSync && localToDevicePaths !== undefined;
+		for (let localToDevicePath of localToDevicePaths) {
```

Comment:
this code will fail in case localToDevicePaths is `undefined`. Based on the check above, it looks like it is possible to have `undefined`, so I would suggest to use 
```TypeScript
_.each(localToDevicePaths, localToDevicePath => {
});
```